### PR TITLE
Fixed option to store users without password, so they are actually stored, and cannot be logged on to.

### PR DESCRIPTION
**Pull request by bert-laverman**
Creating users without a password was already possible, but actually storing them would fail due to a "NOT NULL" constraint on the database. This is now fixed by using a dummy value "nologon" (comparable to the nologon-shell in Linux) that will block any attempt to use the account, as the value is not hashed so no actual password will ever match.

### CODE REVIEW

File: axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java [hunk 1]
```diff
@@ -61,36 +94,43 @@ private User syncUser(String username, String password, Set<UserRole> roles) {
...
-        return syncUser(username, password == null ? null: passwordEncoder.encode(password), roles);
-    }
-
     @Transactional
     public void syncUser(User jpaUser) {
         synchronized (userRepository) {
             if (StringUtils.isEmpty(jpaUser.getPassword())) {
-                jpaUser.setPassword(getPassword(jpaUser.getUserName()));
+                final String password = getPassword(jpaUser.getUserName());
+                jpaUser.setPassword((password == null) ? PWD_NOLOGON : password);
```
Comment (MGathier) [inline axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java:110]: Should it not check for password == PWD_NOLOGON here?

File: axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java [hunk 2]
```diff
@@ -45,14 +55,37 @@ public void deleteUser(String username) {
...
+     * @param password the plaintext password
+     * @param roles    the roles granted to the user
+     * @return the stored information, including the current (hashed) password.
+     */
+    @Transactional
+    public User updateUser(String username, String password, Set<UserRole> roles) {
         synchronized (userRepository) {
             if (StringUtils.isEmpty(password)) {
                 password = getPassword(username);
+                if (password == null) {
```
Comment (MGathier) [inline axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java:84]: password == PWD_NOLOGON?

File: axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java (hunk 1 above)
Comment (bert-laverman) [reply axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java:110]: The current code uses `null` up to the database, where the column is "`NOT NULL`". I kept the usage of null the same, and substitute "nologon" just before storage. This ensures the database constraint stays valid, and rows without a password are visually recognizable.

Comment (bert-laverman) [reply axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java:110]: As discussed: `getPassword(username)` will only return `null` if the user is not found. Will change to use `findUser(username)` to make this more explicit.

File: axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java (hunk 2 above)
Comment (bert-laverman) [reply axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java:84]: As discussed: getPassword(username) will only return null if the user is not found. Will change to use findUser(username) to make this more explicit.

**Review by MGathier — APPROVED** [review]

**Pull request merged**